### PR TITLE
#184651074 add product to the buyer's Wishlist

### PR DESCRIPTION
**Pull request by Pacific911**
#### What does this PR do?
- It adds a product to the buyer's wishlist
- It ensures the logged in user to have products stored in the wishlist

#### Description of Task to be completed?
- a buyer needs to have an account to store his/her wishlist products

#### How should this be manually tested?
- By the use of this endpoint /wishlist//item/${product id}

#### What are the relevant pivotal tracker/Trello stories?
- 184651074

### CODE REVIEW

**Review by mirindi19 — APPROVED** [review]

**Review by Edcreation — APPROVED** [review]

File: src/database/models/wishlist.model.js
Comment (AdrineUWERA) [inline src/database/models/wishlist.model.js:1]: Add the wishlist and user relationship

File: src/helpers/redis.js
Comment (AdrineUWERA) [inline src/helpers/redis.js:1]: you don't need to configure dotenv in this file

**Review by princoo — APPROVED** [review]

**Review by mirindi19 — APPROVED** [review]

**Review by princoo — APPROVED** [review]

**Review by JosephNgabo — APPROVED** [review]

**Review by munezerobagira — CHANGES_REQUESTED** [review]

File: src/database/models/wishlist.model.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    onUpdate: 'CASCADE',
+    onDelete: 'CASCADE',
+  },
+  products: {
+    type: DataTypes.ARRAY(DataTypes.STRING),
+    allowNull: false,
+    defaultValue: [],
+  },
+});
+export default Wishlist;
```
[the remark this answers is not in the document]
Comment (Pacific911) [reply src/database/models/wishlist.model.js]: requested  change Solved

File: src/tests/collection.test.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /* eslint-disable prefer-destructuring */
```
[the remark this answers is not in the document]
Comment (Pacific911) [reply src/tests/collection.test.js]: solved

File: src/database/seeders/20230421014222-wishlist.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    },
+    ['id']
+  );
+  const [resultOne] = await queryInterface.sequelize.query(`
+  SELECT COUNT(*) AS count FROM users WHERE email='testingwishlist@example.com'
+`);
+  // eslint-disable-next-line eqeqeq
+  if (resultOne[0].count == 0) {
+    await queryInterface.bulkInsert('users', [
+      {
```
Comment (JosephNgabo) [inline src/database/seeders/20230421014222-wishlist.js:23]: I think here you can create variable above for this password and then you call them below,

Comment (Pacific911) [reply src/database/seeders/20230421014222-wishlist.js:24]: requested change done 👍

**Review by JosephNgabo — APPROVED** [review]

File: src/database/models/wishlist.model.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import { DataTypes } from 'sequelize';
+import sequelize from '../config/db';
+import User from './user.model';
+
+const Wishlist = sequelize.define('wishlist', {
+  id: {
+    allowNull: false,
+    autoIncrement: false,
```
Comment (princoo) [inline src/database/models/wishlist.model.js:8]: property `autoIncrement` not needed

Comment (Pacific911) [reply src/database/models/wishlist.model.js]: changes done 👍

**Review by princoo — APPROVED** [review]

File: src/controllers/wishes.controller.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        data: wishlist,
+      })
+    );
+});
+
+const removeFromWishlist = asyncWrapper(async (req, res) => {
+  const { pid } = req.params;
+  await wishListServices
+    .removeProductFromWishlist(req.user.id, pid)
+    .then((wishlist) =>
```
Comment (Edcreation) [inline src/controllers/wishes.controller.js:30]: correct this. no need of `await` since you used `.then()`

Comment (Pacific911) [reply src/controllers/wishes.controller.js:30]: 👍

Comment (Pacific911) [reply src/controllers/wishes.controller.js]: Solved the changes 👍

**Review by munezerobagira — APPROVED** [review]

File: src/database/models/wishlist.model.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  },
+});
+Wishlist.belongsTo(User, { onDelete: 'CASCADE' });
+
+User.hasOne(Wishlist, {
+  as: 'wishlistItems',
+  foreignKey: 'userId',
+  onDelete: 'CASCADE',
+  onUpdate: 'CASCADE',
+});
```
[the remark this answers is not in the document]
Comment (Pacific911) [reply src/database/models/wishlist.model.js:26]: It's already there in relationship as i declared it `foreignKey: 'userId'` i think there is no need of writing it above

[the remark this answers is not in the document]
Comment (munezerobagira) [reply src/database/models/wishlist.model.js:26]: ok

**Review by munezerobagira — APPROVED** [review]

**Pull request merged**